fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractCumulativeLoanScheduleGenerator.java

```diff
@@ -388,7 +388,7 @@ private LoanScheduleModel generate(final MathContext mc, final LoanApplicationTe
             }
             ////////////
 
-            if (!isMidTermRescheduling) {
+            if (!isMidTermRescheduling || loanApplicationTerms.getLoanTermVariations().getInterestRateFromInstallment().isEmpty()) {
```

Comment:
any need tp check for null pointer ? in loanApplicationTerms.getLoanTermVariations() ?

Reply:
No. It will never generate the exception because all these variables are initialized when created